### DISCUSSION TITLE
How could I count all the (rgb 149, 0, 133 or #950085) color pixels from an image?

### DISCUSSION THREAD

**Discussion by Alberto-Magno** [ImageMagick/ImageMagick#2075] [Help] [answered]
Hi everyone, this is my fisrt post so I hope I will do it in the correct form and place!!

I want to be able to pick up one color from an image, let's say the purple (rgb 149, 0, 133 or #950085) from an image that certantly contain the color, then from that image I would like to extract all the pixels (or know the number of pixels) that match with the previously color selected.
I have trayed to pick up the color using "grabc" from the terminal in ubuntu but the color returned seems to be an oposite color, so I tryed pick-colour-picker and the grabed color from the original image is correct, but if I convert the image.png to the image.txt (using imagemagick from terminal: convert image.png image.txt) I never found nither the HEX color nor the sRGB in the image.txt file.

Why the color (rgb 149, 0, 133 or #950085) grabed from the picture is not present in the image.txt?
How could I count all the purple pixels (rgb 149, 0, 133 or #950085) from the image?

Thanks in advance!!

**Comment by fmw42** [comment, marked as answer]
Please always provide your ImageMagick version and platform, since syntax may vary.

Post an example image. Some tools that claim to read the image colors via a color picker do not get the proper values. Using txt: from ImageMagick will get the proper colors. So if your color is not there, then that is correct and you probably have a near color.  To get a count of all pixels of a given color, convert all other colors to black and then that color to white. Then use the average of that image to compute the count as follows. Suppose you want to count red and your image has no transparency, then

`convert image -alpha off -fill black +opaque red -fill white -opaque red -format "%[fx:w*h*mean]\n" info:`

For example:

![cyclops1](https://user-images.githubusercontent.com/13458218/83289083-510b8880-a199-11ea-9f6a-5de3672018dd.png)

`convert cyclops1.png -alpha off -fill black +opaque red -fill white -opaque red +write x.png -format "%[fx:w*h*mean]\n" info:`

4

**Reply by Alberto-Magno** [reply]
Hi  fmw 42,

I really appreciate your answer it helps me a lot, you not only gave me a numerical value for the red pixels even a visiual localization for the color looked for.

I am in Ubuntu 18.04 and my ImageMaick is the 6.9.7-4 version.
On the other hand, using your image (cyclops1.png) I tryed to locate the #0A618315CAE1FFFF color, but changed "0A618315CAE1FFFF" instead of "red" in the comand did not work

`convert cyclops1.png -alpha off -fill black +opaque 0A618315CAE1FFFF -fill white -opaque 0A618315CAE1FFFF +write x.png -format "%[fx:w*h*mean]\n" info:`

Is there any way to looking for the color code or convert the color code to a color name?

**Reply by fmw42** [reply]
That is not the way to write hex colors. You need the # so that it is "#0A618315CAE1FFFF" including the quotes for Unix systems. Also how do you know that color is even in the image?

`convert cyclops1.png -alpha off -fill black +opaque "#ff0000" -fill white -opaque "#ff0000" +write x.png -format "%[fx:w*h*mean]\n" info:`

4

**Reply by Alberto-Magno** [reply]
Hooray!!

I am going to describe the entire process for future references ;-)

1. Download cyclops1.png
2. `convert cyclops1.png cyclops1.txt` (here I saw the #0A618315CAE1FFFF color)
3. `convert cyclops1.png -alpha off -fill black +opaque "#0A618315CAE1FFFF" -fill white -opaque "#0A618315CAE1FFFF" +write x.png -format "%[fx:w*h*mean]\n" info:`
1 (1 is the result)

Thank you so much fmw42, I am new on a unix enviroment and newer using ImageMagick, I like both tools!!!!

Thaks again!!

**Reply by fmw42** [reply]
The +write x.png is optional only if you want to see where the color was in the image.
